source/helpers/on-element-removal.ts

```diff
@@ -1,14 +1,22 @@
 import mem from 'mem';
 
 const onElementRemoval = mem(
-	async (element: Element): Promise<void> => (
+	async (element: Element, {signal}: {signal?: AbortSignal} = {}): Promise<void> => (
```

Comment:
The signal is passed in an "options object" to mimic `addEventListener`.

Reply:
I think we can avoid it in some places like this one since it's not a shared npm module.

```suggestion
	async (element: Element, signal?: AbortSignal): Promise<void> => (
```